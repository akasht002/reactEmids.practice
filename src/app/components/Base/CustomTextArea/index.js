import React from 'react';

class CustomTextArea extends React.Component {
    render() {
        return(<div className="form-group">
            <label htmlFor={this.props.name}>{this.props.name}</label> <i className={this.props.mandatory}></i>
            <textarea 
                id={this.props.name}
                required={this.props.required}
                placeholder={this.props.placeholder}
                className={this.props.className}
                rows={this.props.rows}
                maxLength={this.props.maxlength}
                onChange={this.props.textChange}
                value={this.props.value}
                disabled={this.props.disabled}
            />
            <span className="char-limit-block">{`${this.props.maxlength} character max`}</span>
        </div>
        );
    }
}

export default CustomTextArea;