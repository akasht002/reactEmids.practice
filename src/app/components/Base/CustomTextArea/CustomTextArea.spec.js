import React from 'react';	
import CustomTextArea  from './index';	
import renderer from 'react-test-renderer';	

 test('CustomTextArea Component Testing', () => {	
    const component = renderer.create(	
        <CustomTextArea	
            id={1}	
            required={false}	
            placeholder={'TextArea'}	
            className={'TextArea'}	
            rows={20}	
            maxLength={1000}	
            value={'TextArea'}>	
        </CustomTextArea>,	
    );	
    let tree = component.toJSON();	
    expect(tree).toMatchSnapshot();	
});