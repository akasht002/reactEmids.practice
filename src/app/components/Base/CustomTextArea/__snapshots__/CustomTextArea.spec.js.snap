// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTextArea Component Testing 1`] = `
<div
  className="form-group"
>
  <label />
   
  <i />
  <textarea
    className="TextArea"
    placeholder="TextArea"
    required={false}
    rows={20}
    value="TextArea"
  />
</div>
`;
